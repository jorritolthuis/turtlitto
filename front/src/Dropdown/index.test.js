import React from "react";
import Dropdown from ".";
import { shallow } from "enzyme";

describe("Dropdown", () => {
  it("can be changed", () => {
    let value = "test";
    const wrapper = shallow(
      <Dropdown
        currentValue={value}
        values={["test", "test2"]}
        enabled
        onChange={newValue => {
          value = newValue;
        }}
      />
    );
    expect(wrapper).toMatchSnapshot();

    // Check if we can trigger onChange
    expect(value).toBe("test");
    wrapper.find("select").simulate("change", { target: { value: "test2" } });
    expect(value).toBe("test2");
  });
});
