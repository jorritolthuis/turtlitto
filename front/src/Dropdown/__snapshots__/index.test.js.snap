// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Dropdown can be changed 1`] = `
<select
  className={undefined}
  disabled={false}
  onChange={[Function]}
  value="test"
>
  <option>
    test
  </option>
  <option>
    test2
  </option>
</select>
`;
